src/main/java/com/mineinabyss/mobzy/ecs/serializers/MobTypeSerializer.kt

```diff
@@ -0,0 +1,26 @@
+package com.mineinabyss.mobzy.ecs.serializers
+
+import com.mineinabyss.mobzy.mobs.MobType
+import com.mineinabyss.mobzy.registration.MobzyTypes
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+//TODO: Maybe deserialize as a different type? If we use MobType, the deserialized MobType needs to have been
+// registered before any types with references to that MobType.
+object MobTypeSerializer : KSerializer<MobType> {
```

Comment:
Deserialization type will probably change to something else, as MobType shouldn't be relevant to projectiles. Any inherited class from GearyEntityType will also cause registration order issues (referenced entity needs to be registered before the referencing entity), so I'd like to avoid deserializing into GearyEntityType. Any suggestions?

Reply:
Good point, the whole tree of actions under the events component isn't lazily evaluated. The easiest way to lazily do this is just ask for a string and get mob type via the string, as slightly annoying as that is. Doing this properly requires thinking about how to stop circular dependencies and whatnot and that sounds like a lot of pain as well. 

Just make it take a string for now and then 
```kotlin
val entityType by lazy { MobzyTypes[type] }
```